### DISCUSSION TITLE
Can't find the module I created

### DISCUSSION THREAD

**Discussion by kongbair** [pyinstaller/pyinstaller#6363] [Help]
I don't know why. In the 4.2 version, the package can find the module I created in the current directory, but if I upgrade to the 4.7 version, an error will be reported.Such as:ModuleNotFoundError: No module named xxx
Ask for your answers and help, thank you very much!

**Comment by rokm** [comment]
Can you provide a minimal example that reproduces the problem?

**Comment by kongbair** [comment]
Hi,I wrote my example in the markdown file, please check!


空白
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "pyinstaller/pyinstaller"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年11月14日(星期天) 凌晨4:48
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [pyinstaller/pyinstaller] Can't find the module I created (Discussion #6363)





 
Can you provide a minimal example that reproduces the problem?
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by bwoodsend** [reply]
Your attachment hasn't worked at all. Maybe try via the Github WebUI rather than email?

**Reply by kongbair** [reply]
[My example.md](https://github.com/pyinstaller/pyinstaller/files/7533818/My.example.md)

**Reply by bwoodsend** [reply]
Where is this `utils` module? If it's in the same folder as your `main.py` then it should *just work* as is. Otherwise you may add its parent directory as a module search path via the `--pathex` option but a less hacky, more permanent solution would be to [turn your `utils` package into a pip installable setuptools distribution](https://python-packaging.readthedocs.io/en/latest/index.html) which is the recommended way to keep your packages able to find each other - even without PyInstaller.

**Comment by kongbair** [comment]
Due to the internet, the reply was late, so sorry!

**Comment by kongbair** [comment]
it's in the same folder as my&nbsp;main.py




空白
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "pyinstaller/pyinstaller"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年11月14日(星期天) 晚上9:48
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [pyinstaller/pyinstaller] Can't find the module I created (Discussion #6363)





 
Where is this utils module? If it's in the same folder as your main.py then it should just work as is. Otherwise you may add its parent directory as a module search path via the --pathex option but a less hacky, more permanent solution would be to turn your utils package into a pip installable setuptools distribution which is the recommended way to keep your packages able to find each other - even without PyInstaller.
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by rokm** [comment]
So by minimal example, I meant something like this:

```
(venv) C:\Users\Rok\Development\pyi-6363>dir
 Volume in drive C has no label.
 Volume Serial Number is 2EB3-D451

 Directory of C:\Users\Rok\Development\pyi-6363

14/11/2021  19:57    <DIR>          .
14/11/2021  19:57    <DIR>          ..
14/11/2021  19:58                38 main.py
14/11/2021  19:58                50 util.py
               2 File(s)             88 bytes
               2 Dir(s)  49.889.382.400 bytes free
```
where the contents of the files are as follows:

```python
# main.py
import util

util.hello()
``` 
```python
# util.py
def hello():
    print("Hello world!")
```

In this case, PyInstaller 4.7 correctly picks up `util.py` that's next to the entry-point script `main.py`:

```
(venv) C:\Users\Rok\Development\pyi-6363>pyinstaller --version
4.7
(venv) C:\Users\Rok\Development\pyi-6363>pyinstaller --clean --noconfirm main.py
...
(venv) C:\Users\Rok\Development\pyi-6363>dist\main\main.exe
Hello world!
```

Does this correspond to your scenario/setup, or is it different in some way?
